src/ConfigChanges/ConfigFile.js

```diff
@@ -248,16 +267,19 @@ function getIOSProjectname (project_dir) {
         throw new Error(`${msg} in ${project_dir}`);
     }
 
-    return path.basename(matches[0], '.xcodeproj');
+    const projectName = path.basename(matches[0], '.xcodeproj');
+    xcodeprojMap.set(project_dir, projectName);
+    return projectName;
 }
 
 // determine if a plist file is binary
 // i.e. they start with the magic header "bplist"
+// TODO: Remove in next major
 function isBinaryPlist (filename) {
     return readChunk.sync(filename, 0, 6).toString() === 'bplist';
 }
 
 module.exports = ConfigFile;
-module.exports.isBinaryPlist = isBinaryPlist;
+module.exports.isBinaryPlist = util.deprecate(isBinaryPlist, 'isBinaryPlist is deprecated');
 module.exports.getIOSProjectname = getIOSProjectname;
```

Comment:
On the jasmine vs rewire thing.. this might have been a cause of confusion.

The jasmine way worked because it is spying on `ConfigFile`, and the class definition lacks `getIOSProjectname`... but it gets added on this line (because `module.exports === ConfigFile`, so adding properties on `module.exports` is adding properties on `ConfigFile`).

Perhaps out of scope of this PR but it would make more sense if `getIOSProjectname` becomes an actual static method, especially since we already use [ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static#browser_compatibility):

```javascript
class ConfigFile {
    ...

    static getIOSProjectname (project_dir) {
        ...
    }
}
```

Then we won't need that `module.exports.getIOSProjectname = ...` line